src/ert/ensemble_evaluator/monitor_async.py

```diff
@@ -0,0 +1,135 @@
+        self._connection: Optional[WebSocketClientProtocol] = None
+        self._receiver_task: Optional[asyncio.Task[None]] = None
+        self._connected: asyncio.Event = asyncio.Event()
+
+    async def __aenter__(self) -> "MonitorAsync":
+        self._receiver_task = asyncio.create_task(self._receiver())
+        await self._connected.wait()
+        return self
+
+    async def __aexit__(self, exc_type: Any, exc_value: Any, traceback: Any) -> None:
+        if self._connection:
+            await self._connection.close()
+        if self._receiver_task:
+            if not self._receiver_task.done():
+                self._receiver_task.cancel()
+            # we are done and not interested in errors when cancelling
+            await asyncio.gather(
+                self._receiver_task,
+                return_exceptions=True,
+            )
+
+    def get_base_uri(self) -> str:
+        return self._ee_con_info.url
+
+    async def signal_cancel(self) -> None:
+        if not self._connection:
+            return
+        logger.debug(f"monitor-{self._id} asking server to cancel...")
+
+        out_cloudevent = CloudEvent(
+            {
+                "type": identifiers.EVTYPE_EE_USER_CANCEL,
+                "source": f"/ert/monitor/{self._id}",
+                "id": str(uuid.uuid1()),
+            }
+        )
+        await self._connection.send(
+            to_json(out_cloudevent, data_marshaller=evaluator_marshaller)
+        )
+        logger.debug(f"monitor-{self._id} asked server to cancel")
+
+    async def signal_done(self) -> None:
+        if not self._connection:
+            return
+        logger.debug(f"monitor-{self._id} informing server monitor is done...")
+
+        out_cloudevent = CloudEvent(
+            {
+                "type": identifiers.EVTYPE_EE_USER_DONE,
+                "source": f"/ert/monitor/{self._id}",
+                "id": str(uuid.uuid1()),
+            }
+        )
+        await self._connection.send(
+            to_json(out_cloudevent, data_marshaller=evaluator_marshaller)
+        )
+        logger.debug(f"monitor-{self._id} informed server monitor is done")
+
+    async def track(self) -> AsyncGenerator[CloudEvent, None]:
+        while True:
+            event = await self._event_queue.get()
+            yield event
+            self._event_queue.task_done()
+            if event["type"] == identifiers.EVTYPE_EE_TERMINATED:
+                logger.debug(f"monitor-{self._id} client received terminated")
+                break
+
+    async def _receiver(self) -> None:
+        tls: Optional[ssl.SSLContext] = None
+        if self._ee_con_info.cert:
+            tls = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+            tls.load_verify_locations(cadata=self._ee_con_info.cert)
+        headers = Headers()
+        if self._ee_con_info.token:
+            headers["token"] = self._ee_con_info.token
+
+        await wait_for_evaluator(
```

Comment:
Can this function be renamed to `wait_for_evaluator_to_start`?
I assume that is what it does, but I am not too sure.

Reply:
Maybe this should be a separate PR^